packages/flutter_tools/test/src/fakes.dart

```diff
@@ -445,6 +447,152 @@ class FakePub extends Fake implements Pub {
   }) async { }
 }
 
+class FakeFlutterVersion implements FlutterVersion {
```

Comment:
I know you just moved this, but this test class doesn't seem tight enough considering it's not actually a `Fake` and the `FlutterVersion` constructor is calling `git` commands.

Reply:
I agree, thankfully its not widely used so we should probably find a better pattern.

Reply:
Oh nm it's an `implements`, does this work if you `extends Fake`?

Reply:
`FlutterVersion` was one of the most painful to inject into tests because of those `git` calls.
https://github.com/flutter/flutter/blob/666c9509892026b288ec668be47cae741b57e23a/packages/flutter_tools/lib/src/commands/build_ios_framework.dart#L37

Reply:
What I mean is that I agree we should fix this, but I think since this PR didn't actually modify it (and it is still default injected in the testbed groups) I don't think I should expand the scope of this PR to also fix it.

Reply:
I just checked, the tests pass with:
```dart
class TestFeatureFlags extends Fake implements FeatureFlags {
```
```dart
class FakeFlutterVersion extends Fake implements FlutterVersion {
```
> I know you just moved this, but

> I think since this PR didn't actually modify it I don't think I should expand the scope of this PR to also fix it.

Fair enough. I can play with it if I adopt `FakeFlutterVersion` in more places.

Reply:
TestFeatureFlags definitely isn't a fake though, its pretty specifically designed